public/docs/ts/latest/cookbook/production-deploy.jade

```diff
@@ -0,0 +1,316 @@
+  less round trips. 
+  For small files, the overwhelming majority of time it takes to get it is not spent downloading,
+  but rather communicating with the server and coordinating the transfer. 
+  You can read more about resource timings in the 
+  [Chrome DevTools Network Performance page](https://developers.google.com/web/tools/chrome-devtools/network-performance/understanding-resource-timing)   
+
+  ### Smaller files
+  Minification, Uglification and Dead Code Elimination all contribute to reduce the total size
+  of your code to the smallest possible amount of bytes, which will result in a smaller
+  transfer time.
+
+  Tree shaking also reduces the total size but does so by removing whole exports from modules.
+  It was popularized by [Rollup](http://rollupjs.org/) and you can read more about it in 
+  [this post](https://medium.com/@Rich_Harris/tree-shaking-versus-dead-code-elimination-d3765df85c80#.15ih9cyvl).
+
+  ### Template compilation
+  AoT compilation reduces render time by pre-compiling Angular templates that otherwise would be
+  compiled at runtime. 
+  It also reduces payload size since the Angular compiler itself is no longer needed, and nor are
+  directives that are never used in templates. This allows Tree-Shaking to further remove exports.
+
+  You can read more about AoT Compilation in our [dedicated cookbook chapter](aot-compiler.html),
+  where you'll also find instructions on how to use Rollup to do all the optimizations shown here.
+  Another great choice is [Webpack 2](https://webpack.js.org/) together with the 
+  [Angular Ahead-of-Time Webpack Plugin](https://github.com/angular/angular-cli/tree/master/packages/%40ngtools/webpack).
+  
+  You can use any build system you like however, the only important thing is to optimize your build.
+  Whatever build system you choose make sure to automate it so that you can make a production
+  build in a single step.
+
+.l-main-section
+:marked
+  ## Angular configuration
+
+  ### The `base` tag
+
+  There is an important configuration item in the Angular Router that needs to be adjusted for 
+  different environments: 
+  [the `base` tags `href` property](https://angular.io/docs/ts/latest/guide/router.html#!#base-href).
+  This property tells the router how what the app root URL is so that the router can match routes.
+
+  On your development server, your app is most likely served at the root: `http://localhost:3000/`.
+  Here you use `<base href="/">`, since `/` is the root of your app.
+  
+  But depending on the server setup that you use, it can also be served in a subfolder like
+  `http://localhost:3000/myapp/` or `http://www.mysite.com/myapp/`, where `/myapp/` is the subfolder.
+  In this case you should use `<base href="/myapp/">`, since `/myapp/` is the root of your app and 
+  not `/`.
+
+  When your `base` tag is misconfigured your browser console will show several console, from
+  missing files on wrong paths to router errors about missing routes.
```

Comment:
Is there an easy way to set this up so that the same code works in dev and deployment? Like adding an environment variable? Or somehow changing this in a build step?